src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -2283,9 +2304,20 @@ private void ClassifyInsert(SyntaxNode node)
                     case SyntaxKind.TypeParameter:
                     case SyntaxKind.TypeParameterConstraintClause:
                     case SyntaxKind.TypeParameterList:
+                        ReportError(RudeEditKind.Insert);
+                        return;
+
                     case SyntaxKind.Attribute:
                     case SyntaxKind.AttributeList:
-                        ReportError(RudeEditKind.Insert);
+                        // To allow inserting of attributes we need to check if the inserted attribute
+                        // is a pseudo-custom attribute that CLR allows us to change, or if it is a compiler well-know attribute
+                        // that affects the generated IL, so we defer those checks until semantic analysis.
+
+                        // Unless the attribute is a module/assembly attribute
+                        if (node.IsParentKind(SyntaxKind.CompilationUnit) || node.Parent.IsParentKind(SyntaxKind.CompilationUnit))
```

Comment:
>IsParentKind [](http://example.com/codeflow?start=33&length=12)

Is this because we don't have mechanism in the compiler to represent and edit of an AssemblySymbol. If so, let's file an issue to follow up.

Reply:
Yeah, https://github.com/dotnet/roslyn/issues/53310